### PR TITLE
added optional scope to Observable.removeCallback

### PR DESCRIPTION
**Pull request by abow**

### CODE REVIEW

File: src/Tools/babylon.observable.ts [hunk 1]
```diff
@@ -157,11 +157,17 @@
         /**
          * Remove a callback from the Observable object
          * @param callback the callback to remove. If it doesn't belong to this Observable, false will be returned.
+         * @param scope optional scope.  If used only the callbacks with this scope will be removed.
         */
-        public removeCallback(callback: (eventData: T, eventState: EventState) => void): boolean {
+        public removeCallback(callback: (eventData: T, eventState: EventState) => void, scope?: any): boolean {
```
Comment (Hersir88) [inline src/Tools/babylon.observable.ts]: bind() your function is not enough ?

**Comment by abow** [comment]
I could use the remove function, but then my object would need to hold a reference to the observable.

**Comment by abow** [comment]
I mean a reference to the observer.

**Comment by RaananW** [comment]
As I said - I understand the need for that, I just find keeping a reference to the observer a better practice than keeping a reference to a function. If you once use anonymous function,o or bind, or anything that alters the execution of the function (no idea why, but some will), the removeCallback function will not work. 
But notice that I wrote it after approving this, so - no worries :smile:

**Comment by abow** [comment]
My object always has a reference to the function, so it is no additional code.  Thanks

**Pull request merged**